### ISSUE TITLE
"No order for ID x"

### ISSUE THREAD

**Issue by Trolldemorted** [rmbolger/Posh-ACME#192]
These commands:
```
Set-PAServer LE_PROD
New-PACertificate "somedomain", "*.somedomain" -AcceptTOS -Verbose -DnsPlugin Windows -PluginArgs @{WinServer='localhost'}
```
no longer fetch certs and instead output this error:
```
AUSFÜHRLICH: Using directory https://acme-v02.api.letsencrypt.org/directory
AUSFÜHRLICH: POST https://acme-v02.api.letsencrypt.org/acme/acct/ACCOUNTID with -1-byte payload
AUSFÜHRLICH: received 283-byte response of content type application/json
AUSFÜHRLICH: Using account 49888690
AUSFÜHRLICH: POST https://acme-v02.api.letsencrypt.org/acme/order/ACCOUNTID/968464841 with -1-byte payload
No order for ID 968464841
In C:\Program Files\WindowsPowerShell\Modules\Posh-ACME\3.9.0\Private\Invoke-ACME.ps1:148 Zeichen:9
+         throw [AcmeException]::new($acmeError.detail,$acmeError)
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (:) [], AcmeException
    + FullyQualifiedErrorId : No order for ID 968464841
```

I have updated posh-acme with `update-module -name Posh-ACME` but that did not help. Moving my local storage in AppData\Local\Posh-ACME to somewhere else and trying again as a fresh start did not help either.

Did I do anything wrong?

**Comment by rmbolger** [comment]
I haven't confirmed yet, but it seems like Let's Encrypt may have started purging expired orders from their database. Orders generally have a shorter expiration than their associated certificates. I'm getting the same error trying to check order status on a bunch of previous orders as well.

Regardless of why LE is claiming these orders don't exist, you have uncovered a bug that assumes previous orders will never be deleted. So thanks! I'll start working on a fix while I see if I can confirm what LE is doing.

**Comment by rmbolger** [comment]
If I understand where the problem is happening, you should be able to work around the issue for now by using `Remove-PAOrder` to delete the old local copy of the order that matches the same primary domain name.

**Comment by rmbolger** [comment]
Confirmed this is a recent Boulder functionality [change](/letsencrypt/boulder/commit/83aafd18842e093483d6701b92419ca8f7f1855b) via @cpu in this [community forum post](https://community.letsencrypt.org/t/are-old-orders-being-purged-from-boulder/105696).

Just pushed a few fixes that I'll try to wrap into a release later today.

**Comment by rmbolger** [comment]
Fixes released with version 3.10.0
